src/ol/format/wmsgetfeatureinfoformat.js

```diff
@@ -85,6 +95,14 @@ ol.format.WMSGetFeatureInfo.prototype.readFeatures_ = function(node, objectStack
           ol.format.WMSGetFeatureInfo.layerIdentifier_) >= 0,
           'localName of layer node should match layerIdentifier');
 
+      if (this.layerName_) {
+        var layerName = layer.localName.substring(0, layer.localName.length -
```

Comment:
we could also replace ol.format.WMSGetFeatureInfo.layerIdentifier_ with an empty string and compare